bin/prepare.js

```diff
@@ -21,67 +27,107 @@ by creating separate databases for them
+        await updateGlobalEnvFile(domainEnvKey, app.serviceUrl)
     }
 
-    for (const app of apps) {
-        const dbName = `${project}-${app}`
-        const redisName = 5 + appsIndexes[app]
-        const port = 4000 + appsIndexes[app]
-        const sport = 8000 + appsIndexes[app]
-        domains += ` ${app}.${localhostDomain}`
-        const serverUrl = (secure) ? `https://${app}.${localhostDomain}:${sport}` : `http://${localhostDomain}:${port}`
-
-        console.log(`====> prepare '${app}' database`)
-        await createPostgresDatabaseInsideDockerComposeContainerIfNotExists(dbName)
-        await prepareMinimalAppEnv(app, dbName, redisName, port, sport, serverUrl)
-        const migrateResult = await runAppPackageJsonScript(app, 'migrate')
-        if (migrateResult) console.log(`----> '${app}' > migrate ${JSON.stringify(migrateResult)}`)
+    // Step 4. Filter out apps that you don't need, save
+    const filteredApps = filter ? appsWithData.filter(app => filter.includes(app.name)) : appsWithData
+
+    // Step 5. Create missing databases and update apps .env if needed
+    for (const app of filteredApps) {
+        console.log(`====> Preparing ${app.name} databases / .env`)
+        if (app.type === 'KS') {
+            console.log('========> Creating PG DB if not exists')
+            await createPostgresDatabaseInsideDockerComposeContainerIfNotExists(app.pgName)
+            console.log('========> Filling default .env values')
+            const env = {
+                COOKIE_SECRET: `${app.name}-secret`,
+                DATABASE_URL: `${LOCAL_PG_DB_PREFIX}/${app.pgName}`,
+                REDIS_URL:`${LOCAL_REDIS_DB_PREFIX}/${app.redisIndex}`,
+                PORT: String(app.port),
+                SPORT: String(app.sport),
+                SERVER_URL: app.serviceUrl,
+            }
+            await prepareMinimalAppEnv(app.name, env)
+            console.log('========> Running migration script')
+            const migrateResult = await runAppPackageJsonScript(app.name, 'migrate')
+            if (migrateResult) console.log(migrateResult)
+        } else if (app.type === 'Next') {
+            // NOTE: Server url is not filled, since it's per-app logic and probably should just use domains as well
+            console.log('========> Next app does not need databases and default .env values')
+        } else {
+            throw new Error('Unknown app type')
+        }
     }
 
-    for (const app of apps) {
-        const prepareResult = await runAppPackageJsonScript(app, 'prepare')
-        if (prepareResult) console.log(`----> '${app}' > prepare ${JSON.stringify(prepareResult)}`)
+    // Step 6. Run prepare.js script of individual apps according to dependencies-graph
+    console.log('====> Executing prepare script of individual apps')
+    const filterArgs = filteredApps.map(app => `--filter=${app.name}`).join(' ')
+    const { stdout, stderr } = await safeExec(`turbo run prepare ${filterArgs}`)
```

Comment:
It will be nice if we will have a chance to change all commands to `yarn run` or npm

Reply:
Turbo is just an orchestrator. `turbo run <script-name> --filter=<app1> --filter=<app2>` = run `<script-name>` from package.json of `app1` and `app2` (if exists)

Reply:
But under the hood it:
1. Build dependency graph
2. Determines number of available threads
3. Determines which jobs can be run in parallel

Reply:
So, basically it's just `yarn run`, but orchestrated